src/inmanta/deploy/scheduler.py

```diff
@@ -644,6 +782,11 @@ async def _new_version(
         requires: Mapping[ResourceIdStr, Set[ResourceIdStr]],
         up_to_date_resources: Optional[Set[ResourceIdStr]] = None,
         reason: str = "Deploy was triggered because a new version has been released",
+        start_deployment: bool = True,
+        intent_lock_acquired: bool = False,
+        scheduler_lock_acquired: bool = False,
+        *,
+        connection: Optional[asyncpg.connection.Connection] = None,
```

Comment:
As it's implemented now, the scheduler lock lifetime might be interleaved with the transaction lifetime (as in the first db operation in the transaction is within the lock, but the lock is released before the transaction closes). I think this happens to be fine because of what we write to the database both here and at deploy done. But
1. I'm not sufficiently confident of this statement
2. It's a very big 

So I think we have to consider if it's worth it, or if perhaps we need to open a second transaction or move the other writes inside the scheduler lock. If we keep it as is I'd like a very explicit comment here and/or where we open the transaction.

This may also be affected by what happens above this method depending on which parameters are passed (any of the four new parameters). Which ties into the difficulty to reason about that I pointed out in another comment.